workflows/cloud_run_job/main.tf

```diff
@@ -0,0 +1,224 @@
+  project = data.google_project.project.id
+  role    = "roles/workflows.invoker"
+  member  = "serviceAccount:${google_service_account.workflows.email}"
+}
+
+# Grant the Cloud Storage service agent permission to publish Pub/Sub topics
+data "google_storage_project_service_account" "gcs_account" {}
+resource "google_project_iam_member" "pubsubpublisher" {
+  project = data.google_project.project.id
+  role    = "roles/pubsub.publisher"
+  member  = "serviceAccount:${data.google_storage_project_service_account.gcs_account.email_address}"
+}
+
+# [END workflows_terraform_iam]
+
+# [START workflows_terraform_create_bucket]
+# Cloud Storage bucket names must be globally unique
+resource "random_id" "bucket_name_suffix" {
+  byte_length = 4
+}
+
+# Create a Cloud Storage bucket
+resource "google_storage_bucket" "default" {
+  name                        = "input-${data.google_project.project.name}-${random_id.bucket_name_suffix.hex}"
+  location                    = "us-central1"
+  storage_class               = "STANDARD"
+  force_destroy               = false
+  uniform_bucket_level_access = true
+}
+# [END workflows_terraform_create_bucket]
+
+# [START workflows_terraform_create_ar_repo]
+# Create an Artifact Registry repository
+resource "google_artifact_registry_repository" "default" {
+  location      = "us-central1"
+  repository_id = "my-repo"
+  format        = "docker"
+}
+# [END workflows_terraform_create_ar_repo]
+
+# [START workflows_terraform_create_run_job]
+# Create a Cloud Run job
+resource "google_cloud_run_v2_job" "default" {
+  name     = "parallel-job"
+  location = "us-central1"
+
+  template {
+    task_count = 10
+    template {
+      containers {
+        image   = "us-central1-docker.pkg.dev/${data.google_project.project.name}/${google_artifact_registry_repository.default.repository_id}/parallel-job:latest"
+        command = ["python"]
+        args    = ["process.py"]
+        env {
+          name  = "INPUT_BUCKET"
+          value = google_storage_bucket.default.name
+        }
+        env {
+          name  = "INPUT_FILE"
+          value = "input_file.txt"
+        }
+      }
+    }
+  }
+}
+# [END workflows_terraform_create_run_job]
+
+# [START workflows_terraform_create_workflow]
+# Create a workflow
+resource "google_workflows_workflow" "default" {
+  name        = "cloud-run-job-workflow"
+  region      = "us-central1"
+  description = "Workflow that routes a Cloud Storage event and executes a Cloud Run job"
+
+  # Note that $$ is needed for Terraform
+  source_contents = <<EOF
+  main:
+      params: [event]
+      steps:
+          - init:
+              assign:
+                  - project_id: $${sys.get_env("GOOGLE_CLOUD_PROJECT_ID")}
+                  - event_bucket: $${event.data.bucket}
+                  - event_file: $${event.data.name}
+                  - target_bucket: "input-${data.google_project.project.name}-${random_id.bucket_name_suffix.hex}"
```

Comment:
```suggestion
                  - target_bucket: "${google_storage_bucket.default.name}"
```

Reply:
@camiekim This is what I meant by simplification

Reply:
Ah, ok, thanks